### ISSUE TITLE
thumbnail seekbar

### ISSUE THREAD

**Issue by candrapersada** [mpvnet-player/mpv.net#520]
how to use thumbnail seekbar on mpvnet?

**Comment by po5** [comment]
You will need a script like [thumbfast](https://github.com/po5/thumbfast) and a compatible osc script.  
All thumbnail scripts require you to have an installation of regular mpv accessible in PATH to generate thumbnails, even if you use a frontend.  
If you're using the TheAMM script (or a fork of it) and have ffmpeg in PATH, you can also force it to use that instead of mpv for generating thumbnails.

**Comment by candrapersada** [comment]
> You will need a script like [thumbfast](https://github.com/po5/thumbfast) and a compatible osc script.

`[thumbfast] mpv subprocess create failed`

**Comment by po5** [comment]
> All thumbnail scripts require you to have an installation of regular mpv accessible in PATH to generate thumbnails, even if you use a frontend.

**Comment by tsl0922** [comment]
> All thumbnail scripts require you to have an installation of regular mpv accessible in PATH to generate thumbnails, even if you use a frontend.

@po5 This can be implemented in libmpv based application, I've added support for it to [ImPlay](https://github.com/tsl0922/ImPlay) (https://github.com/tsl0922/ImPlay/commit/f548e7387e37d98c4f13b3d120ff0d795e00f4c2), it works very well.

To make it work, you need to change `local mpv_path = "mpv"` to `local mpv_path = "ImPlay"` (not exposed by thumbfast, maybe it can be retrived from mpv (https://github.com/mpv-player/mpv/issues/8918)).

**Comment by stax76** [comment]
In the last commit and auto build is:

- Support for encoding mode and thumbfast.
- For script authors, the following info is available in user-data:  
  user-data/frontend/name=mpv.net  
  user-data/frontend/version=7.0.0.0  
  user-data/frontend/process-path=the process path

**Comment by po5** [comment]
Nice thanks!

**Comment by candrapersada** [comment]
how to use thumbfast in mpvnet?

**Comment by stax76** [comment]
It should work with the new beta version of mpv.net released yesterday, but currently needs a special configuration in thumbfast, which is documented here:

https://github.com/po5/thumbfast#mpv-frontends

**Comment by candrapersada** [comment]
but there is no information about it
Set `mpv_path=mpvnet` in `script-opts/thumbfast.conf` or `script-opts-append=thumbfast-mpv_path=mpvnet` in `mpv.conf`.

**Comment by Sneakpeakcss** [comment]
> `script-opts/thumbfast.conf` works good for this. If you have mpv.net in the Path env var, then you can do `mpv_path=mpvnet`, otherwise you need `mpv_path=<full path>`. Having mpv.net in the Path env var is the better choice.

I don't think you even need to add it in Path env, it's working fine on my end after only changing it to `mpv_path=mpvnet`

**Comment by nkh0472** [comment]
Just a remainer for newcomers who are unfamiliar with using scripts, here's a quick guide on using `Thumbfast`:

1. Open the config folder (Right-click Menu - Settings - Open Config Folder);
2. Inside the configuration folder, create a new folder named `scripts` and another new folder `script-opts`.
3. Download the [thumbfast.lua](https://github.com/po5/thumbfast/blob/master/thumbfast.lua) and [osc.lua](https://github.com/po5/thumbfast/blob/vanilla-osc/player/lua/osc.lua), put them into `scripts`.
4. Download the [thumbfast.conf](https://github.com/po5/thumbfast/blob/master/thumbfast.conf), put it into `script-opts`, and edit it according to above discussion.

> `script-opts/thumbfast.conf` works good for this. If you have mpv.net in the Path env var, then you can do `mpv_path=mpvnet`, otherwise you need `mpv_path=<full path>`. Having mpv.net in the Path env var is the better choice.

6. Restart the mpv.net, then it is ready to use.
